src/value/mod.rs

```diff
@@ -318,6 +381,46 @@ pub(crate) fn str_to_c_str(string: &str) -> *const std::os::raw::c_char {
     unsafe { (*c_str).as_ptr() }
 }
 
+pub(crate) fn naive_date_to_mg_date(input: &NaiveDate) -> *mut bindings::mg_date {
+    let unix_epoch = NaiveDate::from_ymd(1970, 1, 1).num_days_from_ce();
+    unsafe { bindings::mg_date_make((input.num_days_from_ce() - unix_epoch) as i64) }
+}
+
+pub(crate) fn naive_local_time_to_mg_local_time(input: &NaiveTime) -> *mut bindings::mg_local_time {
+    let nsec_in_sec = 1_000_000_000_i64;
+    let hours_ns = (input.hour() as i64) * 60 * 60 * nsec_in_sec;
+    let minutes_ns = (input.minute() as i64) * 60 * nsec_in_sec;
+    let seconds_ns = (input.second() as i64) * nsec_in_sec;
+    let nanoseconds = input.nanosecond() as i64;
+    unsafe { bindings::mg_local_time_make(hours_ns + minutes_ns + seconds_ns + nanoseconds) }
+}
+
+pub(crate) fn naive_local_date_time_to_mg_local_date_time(
+    input: &NaiveDateTime,
+) -> *mut bindings::mg_local_date_time {
+    let unix_epoch = NaiveDate::from_ymd(1970, 1, 1).num_days_from_ce();
+    let days_s = (input.num_days_from_ce() - unix_epoch) as i64 * 24 * 60 * 60;
+    let hours_s = (input.hour() as i64) * 60 * 60;
```

Comment:
I see this `60 * 60` a lot. Same for minutes, etc. Maybe define a function such that you don't have to replicate the logic across different function calls ?

Reply:
You mean, function, e.g. `days2seconds(i64)`, I was thinking about that, but I need a new function for almost every line, I can try and minimize the number of functions... probably the better approach is just to define constants and multiply stuff (`h_in_day`, `sec_in_h`, it's more explicit). What would you like to do? Or, did you think about something else?

Reply:
That's I think minimal in terms of code + quite obvious what is going on :smile:

Reply:
I was thinking, for example: 
 `(input.hour() as i64) * 60 * 60;` is hours converted to seconds. 
` (input.num_days_from_ce() - unix_epoch) as i64 * 24 * 60 * 60;` is days converted to seconds. 

You could write two functions for this (pseudocoded -- I need to learn Rust :P ) 
```
hours_as_seconds(hours as i64) {
  return hours * 60 * 60;
}

days_as_seconds(days as i64) {
  return hours_as_seconds(days * 24);
}
```

And now both this comment and the comment below will use these common functions to convert between units of time

Reply:
And then you can compose similar functions for the lines you replicating unit time conversions. That way, you have a single point to change if you make any mistake that will work on lines automatically

Reply:
Take a look now, I think that approach makes total sense :smile:

Reply:
And it looks so much better! No hard coded constants ;)